### ISSUE TITLE
umask setting not honored

### ISSUE THREAD

**Issue by imba3r** [borgbackup/borg#1859]
Hi, I create daily backups with `--umask 022` command but I keep ending up with files that have wrong permissions:

```
borg@node:/tank/backups/borg$ borg info node::node-2016-11-18
Name: node-2016-11-18
Fingerprint: b9c7b7db34497bae9ba03eef09b9c3a564da8e44fa2693bf9d6030f51fdf4245
Hostname: node
Username: borg
Time (start): Fri, 2016-11-18 04:00:09
Time (end):   Fri, 2016-11-18 04:00:50
Command line: borg create --umask 022 --compression none -v --stats --progress /tank/backups/borg/node::node-2016-11-18 /etc /srv/config /usr/local/bin /tank/photos
Number of files: 81320

                       Original size      Compressed size    Deduplicated size
This archive:              133.63 GB            133.63 GB            110.07 MB
All archives:                1.75 TB              1.75 TB             72.41 GB

                       Unique chunks         Total chunks
Chunk index:                  103850              1833069
```

```
(...)
-rw-r--r-- 1 borg borg  7277083 Nov 18 04:00 16619
-rw-r--r-- 1 borg borg  6374894 Nov 18 04:00 16620
-rw-r--r-- 1 borg borg  5440552 Nov 18 04:00 16621
-rw-r--r-- 1 borg borg  5431334 Nov 18 04:00 16622
-rw-r--r-- 1 borg borg  5347880 Nov 18 04:00 16623
-rw------- 1 borg borg  8138494 Nov 18 04:00 16628
-rw------- 1 borg borg  5304544 Nov 18 04:00 16629
-rw------- 1 borg borg  5286260 Nov 18 04:00 16630
-rw------- 1 borg borg  6178959 Nov 18 04:00 16631
-rw------- 1 borg borg  6006489 Nov 18 04:00 16632
-rw------- 1 borg borg  5850191 Nov 18 04:00 16633
-rw------- 1 borg borg  7242004 Nov 18 04:00 16634
(...)
```

Any idea whats going on here or did I misunderstand the umask option?

**Comment by ThomasWaldmann** [comment]
@imba3r are you sure this was all created using the same client with the same umask setting?

Or are multiple clients with different umask settings using this borg repo?

**Comment by enkore** [comment]
^- umask is needed for any command that modifies the repo (create/delete/prune).

**Comment by imba3r** [comment]
Hi, I didn't set umask for prune, that must have been it. I can't verify now as there is nothing to prune but I will report back tomorrow. Thanks!

**Comment by ThomasWaldmann** [comment]
TODO: document that when using --umask, all repo-modifying borg commands create/delete/prune (on all machines accessing the respective repo) must use consistent --umask values.

**Comment by enkore** [comment]
Fixed by #1863
